packaging/version.py

```diff
@@ -198,7 +198,7 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        match = self._regex.search(str(version))
```

Comment:
That's what make sure that the regex doesn't throw a `TypeError` if `version` is not a string (like, `None`)
I could change to having a check like:
```
if not isinstance(version, str):
    raise InvalidVersion("Invalid version: '{0}' is not a string".format(version))
```
but I should probably look for `buffer` or `unicode` and all "string-like types" too.

The issue is that `Version(1)`, `Version(None)` and `Version(MyWeirdKlass())` should raise `InvalidVersion` and not `TypeError` from the regex.